daft/ai/protocols.py

```diff
@@ -24,3 +25,20 @@ class TextEmbedderDescriptor(Descriptor[TextEmbedder]):
     @abstractmethod
     def get_dimensions(self) -> EmbeddingDimensions:
         """Returns the dimensions of the embeddings produced by the described TextEmbedder."""
+
+
+@runtime_checkable
+class ImageEmbedder(Protocol):
+    """Protocol for image embedding implementations."""
+
+    def embed_image(self, image: list[np.ndarray[Any, Any]]) -> list[Embedding]:
+        """Embeds a batch of images into an embedding vector."""
+        ...
```

Comment:
Please define an Image typing in `typing.py` just like how Embedding is defined. I'm also inclined to say `images` here since it is a list. What do you think?

Reply:
Makes sense